src/LIKWID.jl

```diff
@@ -3,18 +3,21 @@ module LIKWID
 
    const liblikwid = "liblikwid"
    include("marker.jl")
+   include("gpu_marker.jl")
 
    function getProcessorId()
       ccall((:likwid_getProcessorId, liblikwid), Cint, ())
    end
 
    function __init__()
       Marker.init()
+      GPUMarker.init()
```

Comment:
Are these only available conditionally?

Reply:
Yes, they might be. I'll check.

Reply:
When running without CUDA.jl on a node without a GPU the CPU part still works but one gets `CUDA cannot be found and initialized (cuInit failed).`.

To avoid a direct dependence on CUDA.jl I'll go with a Requires.jl based solution if there are no objections.

Reply:
It seems that likwid - at least when compiled with `NVIDIA_INTERFACE=true` - is always looking for CUDA no matter the cli options. Just running `likwid-perfctr -C 0 -g FLOPS_SP -m julia` I get `CUDA runtime library libcudart.so not found.` when the CUDA module on the cluster is not loaded or `CUDA cannot be found and initialized (cuInit failed).` when it is.

Reply:
Right, but what happens if likwid has been compiled without GPU support? Can we still call these functions?

Reply:
No we can't:
```
ERROR: LoadError: InitError: could not load symbol "likwid_gpuMarkerInit":
/p/software/juwels/stages/2020/software/likwid/5.1.0-GCCcore-9.3.0/lib/liblikwid.so: undefined symbol: likwid_gpuMarkerInit
```

Reply:
The nvmon functions and the NvMarkerAPI on top exist only when built with `NVIDIA_INTERFACE=true`. In order to check whether LIKWID was built with Nvidia support on the Lua side, I created a simple `nvSupported()` function that is always there and which basically returns the `NVIDIA_INTERFACE` flag ([with](https://github.com/RRZE-HPC/likwid/blob/master/src/luawid.c#L3489) and [without](https://github.com/RRZE-HPC/likwid/blob/master/src/luawid.c#L3496) Nvidia support).

If the Nvidia support is avalable, `likwid-perfctr` tries to read the GPU topology. This could probably be reduced to the cases where any GPU-related CLI options are defined.

If it does not find the library at the moment, there is a PR for that (https://github.com/RRZE-HPC/likwid/pull/424). Title is a little misleading: it fixes builds with CUDA 11.4 but also with CUDA < 10.0 and I added a two-step lookup for the CUDA libs.

Reply:
> I created a simple `nvSupported()` function that is always there and which basically returns the `NVIDIA_INTERFACE` flag ([with](https://github.com/RRZE-HPC/likwid/blob/master/src/luawid.c?rgh-link-date=2021-09-02T09%3A49%3A06Z#L3489) and [without](https://github.com/RRZE-HPC/likwid/blob/master/src/luawid.c?rgh-link-date=2021-09-02T09%3A49%3A06Z#L3496) Nvidia support).

Great, then I'll use this function instead of checking for the `likwid_gpuMarkerInit` symbol.

> If the Nvidia support is avalable, `likwid-perfctr` tries to read the GPU topology. This could probably be reduced to the cases where any GPU-related CLI options are defined.

I think that would make sense (and was slightly surprising to me).

Reply:
I noted it down as issue: https://github.com/RRZE-HPC/likwid/issues/427